bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java

```
@@ -0,0 +1,314 @@
+
+    builder.readRowSettings()
+        .setRetrySettings(retrySettings)
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** To build BigtableDataSettings#readRowsSettings with default Retry settings. */
+  private static void buildReadRowsSettings(Builder builder, BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+    CallOptionsConfig callOptions = options.getCallOptionsConfig();
+    RetrySettings.Builder retryBuilder = builder.readRowsSettings().getRetrySettings().toBuilder();
+
+    //Timeout for ReadRows
+    Duration rpcTimeout = ofMillis(retryOptions.getReadPartialRowTimeoutMillis());
+    Duration totalTimeout = ofMillis(callOptions.isUseTimeout()
+        ? callOptions.getLongRpcTimeoutMs()
+        : retryOptions.getMaxElapsedBackoffMillis());
+
+    retryBuilder
+        .setInitialRetryDelay(ofMillis(retryOptions.getInitialBackoffMillis()))
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+        .setMaxRetryDelay(ofMillis(MAX_RETRY_TIMEOUT_MS))
+        .setMaxAttempts(retryOptions.getMaxScanTimeoutRetries())
+        .setInitialRpcTimeout(rpcTimeout)
+        .setMaxRpcTimeout(rpcTimeout)
+        .setTotalTimeout(totalTimeout);
+
+    builder.readRowsSettings()
+        .setRetrySettings(retryBuilder.build())
+        .setRetryableCodes(buildRetryCodes(options.getRetryOptions()));
+  }
+
+  /** Creates {@link RetrySettings} for non-streaming idempotent method. */
+  private static RetrySettings buildIdempotentRetrySettings(RetrySettings retrySettings,
+      BigtableOptions options) {
+    RetryOptions retryOptions = options.getRetryOptions();
+    CallOptionsConfig callOptions = options.getCallOptionsConfig();
+    RetrySettings.Builder retryBuilder = retrySettings.toBuilder();
+
+    if (retryOptions.allowRetriesWithoutTimestamp()) {
+      LOG.warn("Retries without Timestamp does not support yet.");
```

This should throw instead of warn